app/script_variant.py

```diff
@@ -0,0 +1,80 @@
+import jautils
+
+from unidecode import unidecode
+from google.appengine.api import memcache
+
+import re
+import logging
+
+def read_dictionary(file_name):
+    dictionary = {}
+    for line in open(file_name, 'r'):
+        kanji, hiragana = line[:-1].split('\t')
+        dictionary[kanji.decode('utf-8')] = hiragana.decode('utf-8')
+    return dictionary
+
+JAPANESE_NAME_DICTIONARY = read_dictionary('japanese_name_dict.txt')
+JAPANESE_LOCATION_DICTIONARY = read_dictionary('jp_location_dict.txt')
+
+def romanize_word(word):
+    """
+    This method romanize all languages by unidecode.
+    If word is hiragana or katakana, it is romanized by jautils.
+    Args:
+        word: should be script varianted
+    Returns:
+        script varianted word
+    """
+    if not word:
+        return word
+
+    if jautils.should_normalize(word):
+        hiragana_word = jautils.normalize(word)
+        return jautils.hiragana_to_romaji(hiragana_word)
+    romanized_word = unidecode(word)
```

Comment:
Why do we have separate codepaths for Japanese and non-Japanese? Is it because of jautils.normalize()? If so, would this work?

if jautils.should_normalize(word):
  word = jautils.normalize(word)
romanized_word = unidecode(word)